src/main/rx/cc2500_redpine.c

```diff
@@ -0,0 +1,227 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "platform.h"
+
+#ifdef USE_RX_REDPINE_SPI
+
+#include "build/build_config.h"
+#include "build/debug.h"
+
+#include "common/maths.h"
+#include "common/utils.h"
+
+#include "config/feature.h"
+
+#include "drivers/adc.h"
+#include "drivers/io.h"
+#include "drivers/io_def.h"
+#include "drivers/io_types.h"
+#include "drivers/resource.h"
+#include "drivers/rx/rx_cc2500.h"
+#include "drivers/rx/rx_spi.h"
+#include "drivers/system.h"
+#include "drivers/time.h"
+
+#include "config/config.h"
+
+#include "io/vtx.h"
+
+#include "pg/rx.h"
+#include "pg/rx_spi.h"
+#include "pg/rx_spi_cc2500.h"
+
+#include "rx/cc2500_common.h"
+#include "rx/cc2500_frsky_common.h"
+#include "rx/cc2500_frsky_shared.h"
+#include "rx/rx_spi_common.h"
+
+#include "sensors/battery.h"
+
+#include "cc2500_redpine.h"
+
+bool redpineFast = true;
+
+#define VTX_STATUS_FRAME 1
+#define SCALE_REDPINE(channelValue) ((2 * channelValue + 2452) / 3)
+#define SWITCH_REDPINE_SPEED_US 2000000
+#define DEFAULT_PACKET_TIME_US 50000
+
+bool isRedpineFast(void)
+{
+    return (redpineFast);
+}
+
+void switchRedpineMode(void)
+{
+    redpineFast = !redpineFast;
+}
+
+#define CHANNEL_START 3
+void redpineSetRcData(uint16_t *rcData, const uint8_t *packet)
+{
+    if (packet[CHANNEL_START] == VTX_STATUS_FRAME && packet[CHANNEL_START + 1] == 0) {
+        vtxSettingsConfigMutable()->band = packet[5] + 1;
```

Comment:
This should only be done when the craft is disarmed. Also, there is no reason to set `rcData` to `PWM_PULSE_MIN` here.